marlowe-dashboard-client/src/MainFrame/View.purs

```diff
@@ -1,20 +1,135 @@
 module MainFrame.View where
 
 import Prelude hiding (div)
+import Contract.View (renderContractSetup, renderContractDetails)
 import Data.Maybe (Maybe(..))
 import Effect.Aff.Class (class MonadAff)
-import Halogen (ComponentHTML)
-import Halogen.HTML (button, div, text)
+import Halogen (ComponentHTML, ClassName(ClassName))
+import Halogen.HTML (a, button, div, div_, header, main, nav, span, text)
 import Halogen.HTML.Events (onClick)
-import MainFrame.Types (Action(..), ChildSlots, State)
+import Halogen.HTML.Properties (classes)
+import MainFrame.Types (Action(..), Card(..), ChildSlots, Screen(..), State, Overlay(..))
+import Material.Icons as Icon
+import Notifications.View (renderNotifications)
+import Template.View (renderTemplateLibrary, renderTemplateDetails)
+import Welcome.View (renderHome, renderContracts)
 
-render ::
-  forall m.
-  MonadAff m =>
-  State ->
-  ComponentHTML Action ChildSlots m
+render :: forall m. MonadAff m => State -> ComponentHTML Action ChildSlots m
```

Comment:
Why do we require `MonadAff` here?

Reply:
ditto in most of functions here

Reply:
David put it in, I guess because we might need it... I've removed it in all the sub-view functions, but let's keep it here for now in case.